pkg/diagnostics/analyze.go

```diff
@@ -0,0 +1,350 @@
+	} else {
+		util.Logger().Printf("NamespaceStore %s type is not supported in analyze\n", namespaceStore.Name)
+	}
+}
+
+// RunAnalyzeResources runs a CLI command
+func RunAnalyzeResources(cmd *cobra.Command, args []string) {
+	collector := collectorInstance(cmd)
+	makeDirForLogs(collector)
+	collector.log.Println("⏳ Running this command will run tests on all your resources")
+	collector.log.Println("Iterating over all BackingStores...")
+	foundBackingStore := analyzeAllBackingStores(cmd, collector)
+	collector.log.Println("Iterating over all NamespaceStore...")
+	foundNamespaceStore := analyzeAllNamespaceStores(cmd, collector)
+	if foundBackingStore || foundNamespaceStore {
+		destDir, _ := cmd.Flags().GetString("dir")
+		collector.ExportDiagnostics(destDir)
+	}
+}
+
+func analyzeAllBackingStores(cmd *cobra.Command, collector *Collector) bool {
+	list := &nbv1.BackingStoreList{
+		TypeMeta: metav1.TypeMeta{Kind: "BackingStoreList"},
+	}
+	if !util.KubeList(list, &client.ListOptions{Namespace: options.Namespace}) {
+		collector.log.Printf("Could not get BackingStoreList.\n")
+		return false
+	}
+	if len(list.Items) == 0 {
+		collector.log.Printf("No BackingStores found.\n")
+		return false
+	}
+
+	foundBackingStore := false
+	for _, backingStore := range list.Items {
+		if checkIfBackingStoreTypeIsSupported(&backingStore) {
+			foundBackingStore = true
+			analyzeBackingstore(cmd, &backingStore, collector)
+		} else {
+			collector.log.Printf("BackingStore %s type is not supported in analyze\n", backingStore.Name)
+		}
+	}
+	return foundBackingStore
+}
+
+func analyzeAllNamespaceStores(cmd *cobra.Command, collector *Collector) bool {
+	list := &nbv1.NamespaceStoreList{
+		TypeMeta: metav1.TypeMeta{Kind: "NamespaceStoreList"},
+	}
+	if !util.KubeList(list, &client.ListOptions{Namespace: options.Namespace}) {
+		collector.log.Printf("Could not get NamespaceStoreList.\n")
+		return false
+	}
+	if len(list.Items) == 0 {
+		collector.log.Printf("No NamespaceStores found.\n")
+		return false
+	}
+
+	foundNamespaceStore := false
+	for _, namespaceStore := range list.Items {
+		if checkIfNamespaceStoreTypeIsSupported(&namespaceStore) {
+			foundNamespaceStore = true
+			analyzeNamespaceStore(cmd, &namespaceStore, collector)
+		} else {
+			collector.log.Printf("NamespaceStore %s type is not supported in analyze\n", namespaceStore.Name)
+		}
+	}
+	return foundNamespaceStore
+}
+
+func checkIfBackingStoreTypeIsSupported(backingStore *nbv1.BackingStore) bool {
+	if util.IsSTSClusterBS(backingStore) || backingStore.Spec.Type == nbv1.StoreTypePVPool {
+		return false
+	}
+	return true
+}
+
+func checkIfNamespaceStoreTypeIsSupported(namespaceStore *nbv1.NamespaceStore) bool {
+	if util.IsSTSClusterNS(namespaceStore) || namespaceStore.Spec.Type == nbv1.NSStoreTypeNSFS {
+		return false
+	}
+	return true
+}
+
+func analyzeBackingstore(cmd *cobra.Command, backingStore *nbv1.BackingStore, collector *Collector) {
+	analyzeResourceJob := loadAnalyzeResourceJob()
```

Comment:
for the job pod, we need to propagate proxy envs (`HTTP_PROXY`, `HTTPS_PROXY` and `NO_PROXY`)  and probably CA env (`NODE_EXTRA_CA_CERTS` @jackyalbo?).
I think it's better to copy it from the noobaa-core STS

Reply:
I think you should, otherwise, you won't be able to check internally supported backingstores like RGW or maybe another NooBaa(which I guess will be something rarely used), as both use self-signed certs signed by OCP, which is added to this list.